### PR TITLE
Fix ifdef DEBUGs to NDEBUG

### PR DESCRIPTION
**Pull request by ZUUL42**
@trilinos/triutils 

## Description

Correcting `ifdef DEBUG` to `NDEBUG` for recent changes in #5153.

## Checklist

- [x] My commit messages mention the appropriate GitHub issue numbers.
- [x] My code follows the code style of the affected package(s).
- [ ] My change requires a change to the documentation.
- [x] I have read the [code contribution guidelines](../blob/master/CONTRIBUTING.md) for this project.
- [x] All new and existing tests passed.
- [x] No new compiler warnings were introduced.
- [ ] These changes break backwards compatibility.

### CODE REVIEW

File: packages/triutils/src/Trilinos_Util_ReadHpc2Epetra.cpp [hunk 1]
```diff
@@ -80,7 +80,7 @@ void Trilinos_Util_ReadHpc2Epetra_internal(
     exit(1);
   }
   int_type numGlobalEquations, total_nnz;
-#ifdef DEBUG
+#ifdef NDEBUG
```
Comment (mhoemmen) [inline packages/triutils/src/Trilinos_Util_ReadHpc2Epetra.cpp]: It's `ifndef NDEBUG`.  `assert` only exists if `NDEBUG` is _not_ defined.

https://en.cppreference.com/w/c/error/assert

Comment (ZUUL42) [reply packages/triutils/src/Trilinos_Util_ReadHpc2Epetra.cpp]: Ahh!
I'm trying to burn through this a bit too fast today.

**Comment by bathmatt** [comment]
Let me know via my SNL email what other patches you need reviewed, you mentioned yesterday that there were other MRs ??

File: packages/triutils/src/Trilinos_Util_ReadHpc2Epetra.cpp [hunk 2]
```diff
@@ -91,12 +91,12 @@ void Trilinos_Util_ReadHpc2Epetra_internal(
   // the warning is easy to fix.
   if(sizeof(int) == sizeof(int_type)) {
     int numGlobalEquations_int, total_nnz_int;
-#ifdef DEBUG
+#ifndef NDEBUG
     cnt =
 #endif
     fscanf(in_file,"%d",&numGlobalEquations_int);
     assert(cnt > 0);
```
Comment (jmgate) [inline packages/triutils/src/Trilinos_Util_ReadHpc2Epetra.cpp]: Doesn't this line need to be inside one of the `#if` blocks?

Comment (mhoemmen) [reply packages/triutils/src/Trilinos_Util_ReadHpc2Epetra.cpp]: @jmgate It's fine -- `assert` has the following definition, according to https://en.cppreference.com/w/cpp/error/assert :
```c++
#ifdef NDEBUG
#define assert(condition) ((void)0)
#else
#define assert(condition) /*implementation defined*/
#endif
```

Comment (bartlettroscoe) [reply packages/triutils/src/Trilinos_Util_ReadHpc2Epetra.cpp]: The macro `assert()` is only defined if the macro `NDEBUG` is **not** defined.

**Comment by bartlettroscoe** [comment]
We need to make one of the Trilinos PR builds set `CMAKE_BUILD_TYPE=DEBUG` which will not define `NDEBUG` and will catch this type of error in the future.  See #5184.

**Comment by ZUUL42** [comment]
> We need to make one of the Trilinos PR builds set `CMAKE_BUILD_TYPE=DEBUG` which will not define `NDEBUG` and will catch this type of error in the future. See #5184.

We were just discussing the possibility and path forward for a PR debug build at the standup this morning. Nothing concrete yet.

File: packages/triutils/src/Trilinos_Util_ReadHpc2Epetra.cpp (hunk 2 above)
Comment (jmgate) [reply packages/triutils/src/Trilinos_Util_ReadHpc2Epetra.cpp]: Does the fact that all other occurrences of `cnt` are wrapped in an `#if` not come into play here?  @mhoemmen, it sounds like you're saying line 98 gets replaced with `((void)0)` so it won't throw this error again?
```
error: ‘cnt’ was not declared in this scope
```

I'm asking Santa for a Trilinos without `#ifdef`s this year.

Comment (mhoemmen) [reply packages/triutils/src/Trilinos_Util_ReadHpc2Epetra.cpp]: @jmgate wrote:
> @mhoemmen, it sounds like you're saying line 98 gets replaced with `((void)0)` so it won't throw this error again?

Yes, that's correct.  `assert` is a macro.  If `NDEBUG` is defined, then the macro's argument does not appear in the compiled source code.  That is, in fact, why the compiler was warning about "unused variables."

> I'm asking Santa for a Trilinos without `#ifdef`s this year.

I ... don't have anything constructive to say about this ;-)

**Pull request merged**